FireboltDotNetSdk.Tests/Integration/SystemEngineTest.cs

```diff
@@ -35,25 +32,25 @@ public void Cleanup()
                 var cursor = Connection.CreateCursor();
                 try
                 {
-                    cursor.Execute($"DROP ENGINE IF EXISTS {EngineName}");
+                    //We first attach the engine to an existing database as it can't be dropped if not attached
+                    cursor.Execute($"ATTACH ENGINE {newEngineName} TO {Database}");
                 }
-                catch (FireboltException) { };
+                catch (System.Exception) { }
+                try
+                {
+                    cursor.Execute($"DROP ENGINE IF EXISTS {newEngineName}");
+                }
+                catch (System.Exception) { }
                 try
                 {
-                    cursor.Execute($"DROP DATABASE IF EXISTS {DatabaseName}");
+                    cursor.Execute($"DROP DATABASE IF EXISTS {newDatabaseName}");
                 }
                 catch (FireboltException) { };
```

Comment:
Should we fail if cleanup is not possible?

Reply:
I don't want to do assertions in the cleanup method as any failure would leave the method (could play with booleans but I want to keep it short and clean). Instead I will copy that line in a test and I'll keep the cleanup method as is